pkg/operator/hostetcdendpointcontroller/members.go

```diff
@@ -41,19 +43,31 @@ func (h *healthyEtcdMemberGetter) GetHealthyEtcdMembers() ([]string, error) {
 	}
 	hostnames := make([]string, 0)
 	for _, m := range member {
-		hostname := getEtcdName(m.PeerURLS)
+		hostname, err := getEtcdName(m.PeerURLS)
+		if err != nil {
+			return nil, err
+		}
 		hostnames = append(hostnames, hostname)
 	}
 	return hostnames, nil
 }
 
-func getEtcdName(peerURLs []string) string {
+func getEtcdName(peerURLs []string) (string, error) {
 	for _, peerURL := range peerURLs {
 		if strings.Contains(peerURL, "etcd-") {
-			return strings.TrimPrefix(strings.Split(peerURLs[0], ".")[0], "https://")
+			return strings.TrimPrefix(strings.Split(peerURL, ".")[0], "https://"), nil
+		}
+		u, err := url.Parse(peerURL)
+		if err != nil {
+			return "", err
+		}
+		host, port, _ := net.SplitHostPort(u.Host)
+		//TODO peer port should be a global constant
+		if IsIP(host) && port == "2380" {
+			return "etcd-bootstrap", nil
 		}
 	}
-	return ""
+	return "", fmt.Errorf("peerURL %s is not properly formatted", peerURLs[0])
```

Comment:
s/peerURL/peerURLs?

Reply:
that was intended as it was in single context `peerURLs[0]`

Reply:
This case will be hit only if the function does not return in the loop right? so it should be something like

`return "", fmt.Errorf("peerURLs %s is not properly formatted or empty", peerURLs)`

Reply:
changed the function a little as we don't need to loop at all as we only handle [0]. this makes the function more explicit and error handling is more clear.